ytmusicapi/mixins/browsing.py

```diff
@@ -836,34 +882,180 @@ def get_song_related(self, browseId: str):
+            
+            Example when `timestamps` is set to `True` and timestamps are available::
+
+                {
+                    "lyrics": [
+                        LyricLine(
+                            text="I was a liar",
+                            start_time=9200,
+                            end_time=10630,
+                            id=1
+                        ),
+                        LyricLine(
+                            text="I gave in to the fire",
+                            start_time=10680,
+                            end_time=12540,
+                            id=2
+                        ),
+                    ],
+                    "source": "Source: LyricFind",
+                    "hasTimestamps": True
+                }
+
+        """
+
+    def get_lyrics(self, browseId: str, timestamps: bool = False) -> Optional[Lyrics|TimedLyrics]:
+        """
+        Returns lyrics of a song or video. When `timestamps` is set, lyrics are returned with
+        timestamps, if available.
+
+        :param browseId: Lyrics browse-id obtained from :py:func:`get_watch_playlist` (startswith `MPLYt`).
+        :param timestamps: Whether to return bare lyrics or lyrics with timestamps, if available.
+        :return: Dictionary with song lyrics or `None`, if no lyrics are found. 
+            The `hasTimestamps`-key determines the format of the data.
+
+        
+            Example when `timestamps` is set to `False`, or not timestamps are available::
+
+                {
+                    "lyrics": "Today is gonna be the day\\nThat they're gonna throw it back to you\\n",
+                    "source": "Source: LyricFind",
+                    "hasTimestamps": False
+                }
+            
+            Example when `timestamps` is set to `True` and timestamps are available::
+
+                {
+                    "lyrics": [
+                        LyricLine(
+                            text="I was a liar",
+                            start_time=9200,
+                            end_time=10630,
+                            id=1
+                        ),
+                        LyricLine(
+                            text="I gave in to the fire",
+                            start_time=10680,
+                            end_time=12540,
+                            id=2
+                        ),
+                    ],
+                    "source": "Source: LyricFind",
+                    "hasTimestamps": True
+                }
+
+        """
+
+        lyrics: dict = {}
         if not browseId:
-            raise YTMusicUserError("Invalid browseId provided. This song might not have lyrics.")
+            raise YTMusicUserError(
+                "Invalid browseId provided. This song might not have lyrics.")
+
+        if timestamps:
+            # change the client to get lyrics with timestamps (mobile only)
+            copied_context_client = self.context["context"]["client"].copy()
```

Comment:
This could be used in more places in the future, so I would suggest moving it to a separate utility function that returns a deepcopy of the context changed to the mobile context (maybe `get_mobile_context` next to initialize_context in helpers.py)

Reply:
Agreed, however I wouldn't deepcopy the context, because someone might want to do weird stuff with references inside self.context. Using _shallow_-copy here, should structually restore everything to how it was before the call. (so I think)

a cool thing would be using a context manager for use `with`, like
```python
with yt.as_mobile():
    ...
# back to desktop
```
([contextlib](http://docs.python.org/library/contextlib.html)) makes a function a context-manager

Reply:
Interesting idea, I like it. But it won't be pretty to wrap the entire function body in a with statement?

And wouldn't it cause problems during parallel execution? If I call a two endpoint with the same `YTMusic` instance at the same time, wouldn't one of them break if you perform a shallow copy?

Reply:
oh yes, I see what you're meaning. I kinda misunderstood your deepcopy, but if you deepcopied the context only for the operation it's totally fine and thread safe. Will do that.